presto-main/src/main/java/com/facebook/presto/operator/unnest/UnnestOperatorBlockUtil.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.unnest;
+
+import static java.lang.String.format;
+
+final class UnnestOperatorBlockUtil
+{
+    private UnnestOperatorBlockUtil() {}
+
+    private static final int DEFAULT_CAPACITY = 64;
+    // See java.util.ArrayList for an explanation
+    static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+
+    // Copied from io.prestosql.spi.block.BlockUtil#calculateNewArraySize
+    static int calculateNewArraySize(int currentSize)
```

Comment:
Maybe we should move /copy
BlockUtil#calculateNewArraySize to Arrays in presto-array. cc @mbasmanova